### PR TITLE
Updates tough due to TUF Spec #40 changes

### LINKED ISSUE #6: Determine if theupdateframework/specification#40 requires implementation changes
https://github.com/theupdateframework/specification/pull/40

### PR DESCRIPTION
**Pull request by ecpullen**
*Issue #, if available:*
Closes #6 

*Description of changes:*
Root metadata is no longer included in snapshot metadata.

*Testing*
All tests perform as expected.

*Additional Testing*
All additional testing uses the following flow:
* Install a version of tuftool
* Create and write a repo to `outdir` using the installed version of tuftool with simple expirations and versions
* Load and write newly created repo from `outdir` with the tough presented in pr to verify backwards compatibility.

Install tuftool 0.1.0
`cargo install tuftool --version 0.1.0`
Create a repo using a sample root.json with simple expirations and versions
`tuftool create \`
`/Users/pullenep/Code/indir \`
`/Users/pullenep/Code/outdir \`
`--root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json \`
`--snapshot-expires 'in 7 days' \`
`--snapshot-version 1 \`
`--timestamp-expires 'in 7 days' \`
`--timestamp-version 1 \`
`--targets-expires 'in 7 days' \`
`--targets-version 1 \`
`--key /Users/pullenep/Code/snakeoil.pem `
Load the repo using the presented tough library
`cargo run --bin tuftool update \`
`--key /Users/pullenep/Code/snakeoil.pem  \`
`--root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json \`
`--metadata-url file:///Users/pullenep/Code/outdir/metadata \`
`--outdir /Users/pullenep/Code/updated \`
`--snapshot-expires 'in 7 days' \`
`--snapshot-version 1 \`
`--timestamp-expires 'in 7 days' \`
`--timestamp-version 1 \`
`--targets-expires 'in 7 days' \`
`--targets-version 1`

Install tuftool 0.1.1
`cargo install tuftool --version 0.1.1`
Create a repo using a sample root.json with simple expirations and versions
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update to load and write the repo
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

Install tuftool 0.2.0
`cargo install tuftool --version 0.2.0`
Create a repo using a sample root.json with simple expirations and versions
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update to load and write the repo
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

Install tuftool 0.3.0
`cargo install tuftool --version 0.3.0`
Create a repo using a sample root.json with simple expirations and versions
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update to load and write the repo
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

Install tuftool 0.4.0
`cargo install tuftool --version 0.4.0`
`tuftool create --outdir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem -t /Users/pullenep/Code/indir`
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tuftool/tests/create_command.rs [hunk 1]
```diff
@@ -95,8 +95,7 @@ fn create_command() {
     // Ensure the snapshot.json file is correct
     assert_eq!(repo.snapshot().signed.version.get(), snapshot_version);
     assert_eq!(repo.snapshot().signed.expires, snapshot_expiration);
-    assert_eq!(repo.snapshot().signed.meta.len(), 2);
-    assert_eq!(repo.snapshot().signed.meta["root.json"].version.get(), 1);
```
Comment (ecpullen) [inline tuftool/tests/create_command.rs:99]: Root metadata is no longer included in snapshot metadata. So snapshot has 1 less entry, and there is no root.json.

File: tough/src/editor/mod.rs [hunk 2]
```diff
@@ -333,11 +333,9 @@ impl RepositoryEditor {
 
         // Snapshot stores metadata about targets and root
         let targets_meta = Self::snapshot_meta(signed_targets);
-        let root_meta = Self::snapshot_meta(&self.signed_root);
         snapshot
             .meta
             .insert("targets.json".to_owned(), targets_meta);
-        snapshot.meta.insert("root.json".to_owned(), root_meta);
```
Comment (ecpullen) [inline tough/src/editor/mod.rs:340]: Do not include root metadata in snapshot metadata.

**Review by webern — APPROVED** [review]
LGTM, assuming any additional testing that we decide on is done before merge.

**Comment by ecpullen** [comment]
Additional Testing
`cargo install tuftool --version 0.1.0`
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

`cargo install tuftool --version 0.1.1`
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

`cargo install tuftool --version 0.2.0`
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

`cargo install tuftool --version 0.3.0`
`tuftool create /Users/pullenep/Code/indir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem `
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

`cargo install tuftool --version 0.4.0`
`tuftool create --outdir /Users/pullenep/Code/outdir --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1 --key /Users/pullenep/Code/snakeoil.pem -t /Users/pullenep/Code/indir`
Using proposed update
`cargo run --bin tuftool update --key /Users/pullenep/Code/snakeoil.pem  --root /Users/pullenep/Code/Tough/tough/tough/tests/data/simple-rsa/root.json --metadata-url file:///Users/pullenep/Code/outdir/metadata --outdir /Users/pullenep/Code/updated --snapshot-expires 'in 7 days' --snapshot-version 1 --timestamp-expires 'in 7 days' --timestamp-version 1 --targets-expires 'in 7 days' --targets-version 1`

**Review by iliana — APPROVED** [review]

**Pull request merged**